src/NeoSharp.Core/Blockchain/Blockchain.cs

```diff
@@ -214,10 +224,11 @@ public UInt256 GetNextBlockHash(UInt256 hash)
         /// <param name="blockHeaders"></param>
         public void AddBlockHeaders(IEnumerable<BlockHeader> blockHeaders)
         {
-            // TODO: hook up persistence here
-            if (blockHeaders.Any())
+            // TODO: finish this logic
+
+            foreach (var header in blockHeaders)
             {
-                LastBlockHeader = blockHeaders.OrderBy(h => h.Index).Last();
+                _neoSharpRepository.AddBlockHeader(header);
```

Comment:
We still need to update LastBlockHeader

Reply:
You are right, and i think that neo store BlockHeaderBase, not BlockHeader (without TX hashes) ... i need to review this